jdaviz/configs/imviz/helper.py

```diff
@@ -589,7 +577,8 @@ def link_image_data(app, link_type='pixels', wcs_fallback_scheme=None, wcs_use_a
     if link_type == 'pixels' and old_link_type == 'wcs':
         # if changing from WCS to pixel linking, set bottom image data
         # layer as reference data in all viewers:
-        refdata = get_bottom_layer(app._jdaviz_helper.default_viewer)
+        refdata = app.data_collection[0]
```

Comment:
The revision here would correctly simplify the behavior if the entries in the data collection are never changed.

If a user deletes the zeroth data entry, the new zeroth entry may be a WCS-only layer. In that case, the revisions in this diff will use the WCS-only layer as reference data, but not link the data to the WCS-only layer (see your line 605). This will be very confusing for future us.

Reply:
Oh, dear. Do we allow people to delete everything now? Lemme ponder. Thanks for pointing that out.